### ISSUE TITLE
Cannot use generated Subscription types with NestJS resolver

### ISSUE THREAD

**Issue by jon301** [dotansimha/graphql-code-generator#698]
**Describe the bug**

I cannot find a way to use the generated subscription resolver type in a NestJS app 

Given the following schema :
```graphql
type Mutation {
  signIn(email: String!, password: String!): String
}

type Subscription {
  userSignedIn: User
}
```

I get the following generated types :
```typescript
export interface SignInMutationArgs {
  email: string;
  password: string;
}

export namespace MutationResolvers {
  export interface Resolvers<Context = any> {
    signIn?: SignInResolver<string | null, any, Context>;
  }

  export type SignInResolver<
    R = string | null,
    Parent = any,
    Context = any
  > = Resolver<R, Parent, Context, SignInArgs>;
}

export namespace SubscriptionResolvers {
  export interface Resolvers<Context = any> {
    userSignedIn?: UserSignedInResolver<User | null, any, Context>;
  }

  export type UserSignedInResolver<
    R = User | null,
    Parent = any,
    Context = any
  > = Resolver<R, Parent, Context>;
}
```

Then I can create a handy union type of all resolvers :
```typescript
export interface MyAppResolvers extends MutationResolvers.Resolvers, SubscriptionResolvers.Resolvers {};
```

Finally, my resolvers look like this :
```typescript
import { Args, Mutation, Resolver, Subscription } from '@nestjs/graphql';
import { PubSub } from 'apollo-server-express';

import * as GQL from 'src/generated.ts';

const pubSub = new PubSub();

@Resolver()
export class AuthResolver implements GQL.MyAppResolvers {
  @Mutation()
  async signIn(@Args() args: GQL.SignInMutationArgs) {
    // ...
  }

  @Subscription()
  userSignedIn() {
    return {
      subscribe: () => pubSub.asyncIterator('userSignedIn'),
    };
  }
}
```

Then I get a typescript error because `userSignedIn()` resolver does not return a `User` but a function with a `subscribe` method :
```
error TS2416: Property 'userSignedIn' in type 'AuthResolver' is not assignable to the same property in base type 'AppResolver'.
  Type '() => Promise<{ subscribe: () => AsyncIterator<{}>; }>' is not assignable to type 'Resolver<User, any, any, any>'.
    Type 'Promise<{ subscribe: () => AsyncIterator<{}>; }>' is not assignable to type 'User | Promise<User>'.
      Type 'Promise<{ subscribe: () => AsyncIterator<{}>; }>' is not assignable to type 'Promise<User>'.
        Type '{ subscribe: () => AsyncIterator<{}>; }' is not assignable to type 'User'.
          Property 'id' is missing in type '{ subscribe: () => AsyncIterator<{}>; }'.

   async userSignedIn() {
           ~~~~~~~~~~~~
```

I'm quite new with GraphQL here, maybe I should not use a union type of resolvers ?
If so, how can I strongly type my resolver class ?

**Comment by dotansimha** [comment]
@kamilkisiela

**Comment by kamilkisiela** [comment]
@dotansimha #719

**Comment by dotansimha** [comment]
fixed in 0.13.0. @jon301
